core/indigo-core/graph/graph.h

```diff
@@ -125,18 +125,18 @@ namespace indigo
         }
     };
 
-    class GraphMetaObject
+    class CycleBasis;
+
+    class MetaObject
     {
     public:
-        GraphMetaObject(int class_id) : _class_id(class_id)
+        MetaObject(int class_id) : _class_id(class_id)
         {
         }
-        int _class_id;
-        virtual GraphMetaObject* clone() const = 0;
+        uint32_t _class_id;
```

Comment:
It makes sense to keep coordinated types of constructor parameter and data member.

Reply:
fixed